parachain/pallets/commitments/src/lib.rs

```diff
@@ -98,21 +92,31 @@ decl_module! {
 
 impl<T: Config> Module<T> {
 
-	fn offchain_key(hash: <T as Config>::Hash) -> Vec<u8> {
+	fn offchain_key(hash: H256) -> Vec<u8> {
 		(T::INDEXING_PREFIX, hash).encode()
 	}
 
 	// TODO: return proper weight
 	fn commit() -> Weight {
-		let messages: Vec<Message> = <Self as Store>::MessageQueue::take();
+		let mut weight: Weight = 0;
+
+		for channel_id in ChannelId::into_enum_iter() {
+			weight += Self::commit_for_channel(channel_id);
+		}
+
+		weight
+	}
+
+	fn commit_for_channel(channel_id: ChannelId) -> Weight {
+		let messages: Vec<Message> = <Self as Store>::MessageQueues::take(channel_id);
 		if messages.len() == 0 {
 			return 0
 		}
 
 		let commitment = Self::encode_commitment(&messages);
 		let commitment_hash = <T as Config>::Hashing::hash(&commitment);
 
-		let digest_item = AuxiliaryDigestItem::CommitmentHash(commitment_hash.clone()).into();
+		let digest_item = AuxiliaryDigestItem::Commitment(ChannelId::Basic, commitment_hash.clone()).into();
```

Comment:
not sure what this is diong but looks like its using a hardcoded ChannelId::Basic rather than the actual channel_id?

Reply:
Oopsie!, will fix :)